bootstrap/kubeadm/types/v1beta1/types.go

```diff
@@ -41,28 +43,34 @@ type InitConfiguration struct {
 	// configuration object lets you customize what IP/DNS name and port the local API server advertises it's accessible
 	// on. By default, kubeadm tries to auto-detect the IP of the default interface and use that, but in case that process
 	// fails you may set the desired value here.
+	// +optional
 	LocalAPIEndpoint APIEndpoint `json:"localAPIEndpoint,omitempty"`
 }
 
+func (ic InitConfiguration) MarshalJSON() ([]byte, error) {
+	return marshalJSONOmittingEmptyStructs(ic)
+}
```

Comment:
We shouldn't need this I think, rather when we're adding `omitempty`, we should have a pointer